### PR TITLE
Gives keycard scanners more things that they can do

### PR DESCRIPTION
**Pull request by jakmak6**
This PR adds more access lifting to the head of staff keycard scanner. There are currently 6 new options on the keycard scanner. 

Grant Emergency Mining Base Access
Revoke Emergency Mining Base Access
Grant Emergency Xenoarchaeology Base Access
Revoke Emergency Xenoarchaeology Base Access
Remove Crew Weapon Restrictions
Re Enable Crew Weapon Restrictions

The new function does what it says on the tin. Removing the mining base/xenoarch base access removes the access check for mining/xenoarch access on airlocks/windoors/shuttle consoles.

The remove crew weapon restrictions removes the access check for armory on all crates, airlocks, windoors, and lockboxes(WIP).

99% tested

Issues. 
With the xenoarch access grant it gives access to all normal science and RnD access doors since there isnt a regular xenoarch access and the research shuttle/xenoarch base just uses science/RnD access.
The armory access lift removes access on the brig cell windoors. This is because the box station armory has several windoors with the same access as the brig cell doors
Lockbox access removeal is still WIP.
The new options on the keycard scanner are misaligned. https://puu.sh/yER54/bb23b05ac9.png

### CODE REVIEW

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Adding all these global variables and `allowed()` overrides on airlocks is getting ridiculous. Make it into a  generic global list of access levels that doors should ignore, and add/remove from this list for these buttons.

EDIT: The lockbox hardcode is probably still fine though.

File: code/modules/security levels/keycard authentication.dm [hunk 1]
```diff
@@ -88,6 +88,15 @@ var/global/list/obj/machinery/keycard_auth/authenticators = list()
 		dat += {"<li><A href='?src=\ref[src];triggerevent=Grant Emergency Maintenance Access'>Grant Emergency Maintenance Access</A></li>
 			<li><A href='?src=\ref[src];triggerevent=Revoke Emergency Maintenance Access'>Revoke Emergency Maintenance Access</A></li>
 			</ul>"}
+		dat += {"<li><A href='?src=\ref[src];triggerevent=Grant Emergency Mining Base Access'>Grant Emergency Mining Base Access</A></li>
+			<li><A href='?src=\ref[src];triggerevent=Revoke Emergency Mining Base Access'>Revoke Emergency Mining Base Access</A></li>
+			</ul>"}
```
Comment (ghost) [inline code/modules/security levels/keycard authentication.dm:93]: Remove /uls except the last

**Comment by Blithering** [comment]
A setting to allow security into areas would be a neat use

**Comment by Arthurdentist** [comment]
I'd change the name "Xenoarchaeology Base" and "Mining Base" to just "Science Station" and "Mining Station", but that is just me.

**Comment by jakmak6** [comment]
@Blithering i didnt want to add to much to it to prevent it from being to OP.

@Arthurdentist still trying to figure out what sounds best for the new access elevation things

**Pull request closed without merging**